pkgs/tools/system/osquery/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, cmake
+, fetchFromGitHub
+, git
+, llvmPackages
+, nixosTests
+, overrideCC
+, perl
+, python3
+, stdenv
+, fetchurl
+}:
+
+let
+  buildStdenv = overrideCC stdenv llvmPackages.clangUseLLVM;
+  opensslArchive =
+    let
+      # https://github.com/osquery/osquery/blob/877d5e69ab97e15800b5c5128b3de2cf6f322942/libraries/cmake/formula/openssl/CMakeLists.txt#L3-L4.
+      version = "1.1.1q";
+      sha256 = "d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca";
+    in
+    fetchurl {
+      inherit sha256;
+      url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+    };
+in
+buildStdenv.mkDerivation rec {
+  pname = "osquery";
+  version = "5.5.1";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    fetchSubmodules = true;
+    sha256 = "sha256-Q6PQVnBjAjAlR725fyny+RhQFUNwxWGjLDuS5p9JKlU=";
+  };
+
+  patches = [
+    ./Remove-git-reset.patch
+    ./Use-locale.h-instead-of-removed-xlocale.h-header.patch
+    ./Remove-circular-definition-of-AUDIT_FILTER_EXCLUDE.patch
+    # For current state of compilation against glibc in the clangWithLLVM toolchain, refer to the upstream issue in https://github.com/osquery/osquery/issues/7823.
+    ./Remove-system-controls-table.patch
+  ];
+
+
+  buildInputs = [
+    llvmPackages.libunwind
+  ];
+  nativeBuildInputs = [
+    cmake
+    git
```

Comment:
Can we avoid git?

Reply:
Seems like it's primarily used for version detection here: https://github.com/osquery/osquery/blob/5.5.1/cmake/options.cmake#L8

Overriding the CMake variable might solve this.

Reply:
The version is already specified as a Cmake variable.

I just tried to remove git and git is not only used for the version. The build fails on an finding a rpmlib (this doesn't seem to be optionnal):
```
CMake Error at /nix/store/f8inixjb36rcxvjwsghqlqqxgbrdrvly-cmake-3.26.4/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Git (missing: GIT_EXECUTABLE)
Call Stack (most recent call first):
  /nix/store/f8inixjb36rcxvjwsghqlqqxgbrdrvly-cmake-3.26.4/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE
)
  /nix/store/f8inixjb36rcxvjwsghqlqqxgbrdrvly-cmake-3.26.4/share/cmake-3.26/Modules/FindGit.cmake:128 (find_package_handle_standard_args)
  libraries/cmake/source/modules/utils.cmake:77 (find_package)
  libraries/cmake/source/modules/utils.cmake:154 (patchSubmoduleSourceCode)
  libraries/cmake/source/modules/Findlibrpm.cmake:10 (importSourceSubmodule)
  CMakeLists.txt:161 (find_package)
  CMakeLists.txt:57 (importLibraries)
  CMakeLists.txt:211 (main)
```